### ISSUE TITLE
Group conv with a large kernel_size causes `RuntimeError: label is redefined`

### ISSUE THREAD

**Issue by kan-bayashi** [pytorch/pytorch#42446]
## 🐛 Bug

In pytorch 1.6, group conv with a large kernel_size causes `RuntimeError: label is redefined`.
Pytorch 1.0.1 - 1.5.1 works without any errors.

## To Reproduce

```python
import torch
import torch.nn.functional as F


class CNN(torch.nn.Module):
    def __init__(self):
        super(CNN, self).__init__()
        self.layers = torch.nn.ModuleList()

        # first layer is needed to reproduce error
        self.layers += [
            torch.nn.Conv1d(1, 16, 1),
        ]
        self.layers += [
            torch.nn.Conv1d(
                16, 16,
                # kernel_size = 19 is OK, but 21 causes error
                kernel_size=21,
                # without group, no error
                groups=2,
            ),
        ]

    def forward(self, x):
        for f in self.layers:
            x = f(x)
        return x


cnn = CNN()
y = torch.randn(1, 1, 128)
y = cnn(y)
loss = F.mse_loss(y, torch.ones_like(y))
loss.backward()
```

This causes the following error:
```python
Traceback (most recent call last):
  File "test.py", line 34, in <module>
    loss.backward()
  File "/work5/t_hayashi/work/ParallelWaveGAN/tools/venv/lib/python3.6/site-packages/torch/tensor.py", line 185, in backward
    torch.autograd.backward(self, gradient, retain_graph, create_graph)
  File "/work5/t_hayashi/work/ParallelWaveGAN/tools/venv/lib/python3.6/site-packages/torch/autograd/__init__.py", line 127, in backward
    allow_unreachable=True)  # allow_unreachable flag
RuntimeError: label is redefined
```

## Expected behavior

We can call `backward()` without the error.

## Environment

PyTorch version: 1.6.0+cpu
Is debug build: No
CUDA used to build PyTorch: Could not collect

OS: Ubuntu 16.04.6 LTS
GCC version: (Ubuntu 5.4.0-6ubuntu1~16.04.12) 5.4.0 20160609
CMake version: version 3.5.1

Python version: 3.6
Is CUDA available: No
CUDA runtime version: 10.0.130
GPU models and configuration:
GPU 0: TITAN V
GPU 1: TITAN V
GPU 2: TITAN V
GPU 3: TITAN V

Nvidia driver version: 440.64.00
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.5
/usr/local/cuda-10.0/targets/x86_64-linux/lib/libcudnn.so.7.6.2
/usr/local/cuda-10.1/targets/x86_64-linux/lib/libcudnn.so.7.6.5

Versions of relevant libraries:
[pip3] numpy==1.18.4
[pip3] torch==1.6.0+cpu
[conda] Could not collect

## Additional context

I tested with CPU version.
If you want me to provide additional information, please let me know.

cc @gujinghui @PenghuiCheng @XiaobingSuper @jianyuh @VitalyFedyunin

**Comment by agolynski** [comment]
Seems like the error is coming from mkl dnn lib. just make sure: you are using mkl dnn in your installation?

**Comment by vpirogov** [comment]
Thank you for the report @kan-bayashi. I confirm that this is a oneDNN issue affecting systems with Intel AVX512 support.

**Comment by kan-bayashi** [comment]
I checked the cases with the following installations:
- `conda install pytorch=1.6 cudatoolkit=10.2 -c pytorch`
- `pip install torch==1.6.0+cpu -f https://download.pytorch.org/whl/torch_stable.html`
This error happens in both cases. But if we run to gpu, no error happens.

**Comment by vpirogov** [comment]
The issue is fixed in [oneDNN v1.6.1|https://github.com/oneapi-src/oneDNN/releases/tag/v1.6.1].
